packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -385,91 +408,561 @@ class ListTile extends StatelessWidget {
+      : 0.0;
+    return leadingWidth
+      + math.max(_minWidth(title, height), _minWidth(subtitle, height))
+      + _maxWidth(trailing, height);
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    final double leadingWidth = leading != null
+      ? math.max(leading.getMaxIntrinsicWidth(height), _kMinLeadingWidth) + _kLeadingTitleGap
+      : 0.0;
+    return leadingWidth
+      + math.max(_maxWidth(title, height), _maxWidth(subtitle, height))
+      + _maxWidth(trailing, height);
+  }
+
+  double get _defaultTileHeight {
+    final bool hasSubtitle = subtitle != null;
+    final bool isTwoLine = !isThreeLine && hasSubtitle;
+    final bool isOneLine = !isThreeLine && !hasSubtitle;
+
+    if (isOneLine)
+      return isDense ? 48.0 : 56.0;
+    else if (isTwoLine)
+      return isDense ? 64.0 : 72.0;
+    else
+      return isDense ? 76.0 : 88.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return math.max(
+      _defaultTileHeight,
+      title.getMinIntrinsicHeight(width) + (subtitle?.getMinIntrinsicHeight(width) ?? 0.0)
+    );
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return computeMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    assert(title != null);
+    final BoxParentData parentData = title.parentData;
+    return parentData.offset.dy + title.getDistanceToBaseline(TextBaseline.alphabetic);
+  }
+
+  static double _boxBaseline(RenderBox box) {
+    return box.getDistanceToBaseline(TextBaseline.alphabetic);
+  }
+
+  static Size _layoutBox(RenderBox box, BoxConstraints constraints) {
+    if (box == null)
+      return Size.zero;
+    box.layout(constraints, parentUsesSize: true);
+    return box.size;
+  }
+
+  static void _positionBox(RenderBox box, Offset offset) {
+    final BoxParentData parentData = box.parentData;
+    parentData.offset = offset;
+  }
+
+  // All of the dimensions below were taken from the Material Design spec:
+  // https://material.io/design/components/lists.html#specs
+  @override
+  void performLayout() {
+    final bool hasLeading = leading != null;
+    final bool hasSubtitle = subtitle != null;
+    final bool hasTrailing = trailing != null;
+    final bool isTwoLine = !isThreeLine && hasSubtitle;
+    final bool isOneLine = !isThreeLine && !hasSubtitle;
+    final BoxConstraints looseConstraints = constraints.loosen();
+
+    final double tileWidth = looseConstraints.maxWidth;
+    final Size leadingSize = _layoutBox(leading, looseConstraints);
+    final Size trailingSize = _layoutBox(trailing, looseConstraints);
+
+    final double titleStart = hasLeading ? leadingSize.width <= 40.0 ? 56.0 : leadingSize.width + 16.0 : 0.0;
+    final BoxConstraints textConstraints = looseConstraints.tighten(
+      width: tileWidth - titleStart - (hasTrailing ? trailingSize.width + 16.0 : 0.0),
```

Comment:
Should that 16.0 also be _leadingTitleGap? (in which case it should probably be called _titleGap, or add a _trailingTitleGap).

Reply:
Good point. I used _kTitleGap